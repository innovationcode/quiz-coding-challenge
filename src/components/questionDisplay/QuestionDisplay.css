.questionDisplay {
  width: 55vw;
  text-align: left;
  margin: 10% 20%;
  margin-top: 5%;
  height: 100vh;
}

.question {
  border: 1px solid green;
  border-radius: 5px;
  padding: 5px;
  padding-left: 8px;
  box-shadow: 0 0 5px #999;
  display: flex;
  font-family: "Roboto";
  font-weight: bold;
  background-color: azure;
  margin: 9px 0px;
}

.ans-option-button {
  cursor: pointer;
  margin: 14px 0px;
  border: 1px solid green;
  width: 35vw;
  border-radius: 5px;
  padding: 5px;
  padding-left: 8px;
  box-shadow: 0 0 5px #999;
  display: flex;
  font-family: "Roboto";
  font-weight: bold;
  background-color: white;
  outline-style: none;
}

.selected {
  background-color: burlywood;
}

.ans-option-button:hover {
  background-color: burlywood;
}

.next-button {
  width: 100px;
  outline-style: none;
  cursor: pointer;
  display: inline-block;
  padding: 3px;
  outline-style: none;
  margin: 5% 0;
  border-radius: 3px;
}

.next-button:hover {
  background-color: burlywood;
}
