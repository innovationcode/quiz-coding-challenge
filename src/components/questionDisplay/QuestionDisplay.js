import React from 'react';
import './QuestionDisplay.css';


function QuestionDisplay(props) {
      //console.log("Question in QuestionDisplay component  - ", props.ansOptions);

      return (
            <div className= 'questionDisplay'>
                  <span 
                        style={{ fontSize: "16px", fontWeight: "400" }}>
                        {`Question ${props.currentIndex + 1} of ${props.length}`}
                  </span>

                  <h3 className = "question">{props.question}</h3>

                  <p style = {{paddingTop: '12px', paddingBottom:'6px'}}>The options are ...</p>
                  {
                        props.ansOptions ? (props.ansOptions.map((option, index) => {
                             return <button key = {index}
                                          // className = "ans-option-button" 
                                          className = {`ans-option-button ${props.userAnswer === option? "selected" : null}`}
                                          onClick = {() => props.checkAnswer(option)}
                                    >
                                          {option}
                                    </button>
                        })) : <h1>Loading ...</h1>
                  }

                  {props.currentIndex < props.length - 1 && (
                        <button
                              className = "next-button"
                              onClick={props.nextQuestionHandler}
                        >
                              Next
                        </button>
                  )}

                  {props.currentIndex === props.length - 1 && (
                        <button 
                              className = "next-button"
                              onClick={props.finishHandler}
                        >
                              Finish
                        </button>
                  )}
            </div>
      )
}

export default QuestionDisplay;

