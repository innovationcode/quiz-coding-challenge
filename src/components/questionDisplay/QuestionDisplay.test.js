import React from 'react';
import { shallow } from 'enzyme';

import QuestionDisplay from './QuestionDisplay.js'; //component to be tested.

describe('Testing QuestionDisplay component', () => {

      it('should render correctly in "debug" mode', () => {
            const wrapper = shallow(<QuestionDisplay debug />);
            expect(wrapper).toMatchSnapshot();
      });

      it('shold test question display...', () => {
            const wrapper = shallow(<QuestionDisplay />);
            expect(wrapper.find('.question').text()).toBe('');
      });
      
});