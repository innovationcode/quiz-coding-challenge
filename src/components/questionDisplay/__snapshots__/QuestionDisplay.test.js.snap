// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing QuestionDisplay component should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
